test/es-module/test-esm-dynamic-import-mutating-fs.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+
+const assert = require('node:assert');
+const fs = require('node:fs/promises');
+const { pathToFileURL } = require('node:url');
+
+tmpdir.refresh();
+const tmpDir = pathToFileURL(tmpdir.path);
+
+const target = new URL(`./${Math.random()}.mjs`, tmpDir);
+
+(async () => {
+
+  await assert.rejects(import(target), { code: 'ERR_MODULE_NOT_FOUND' });
+
+  await fs.writeFile(target, 'export default "actual target"\n');
+
+  const moduleRecord = await import(target);
+
+  await fs.rm(target);
+
+  assert.strictEqual(await import(target), moduleRecord);
```

Comment:
To ensure that we’re comparing against what was exported, and not inadvertently against an empty object or an error object or something else.

```suggestion
  await import(target);

  await fs.rm(target);

  assert.strictEqual(await import(target), 'actual target');
```

Reply:
No! The spec says:

> If this operation is called multiple times with the same (referrer, specifier) pair and it performs FinishLoadingImportedModule(referrer, specifier, payload, result) where result is a normal completion, then it must perform FinishLoadingImportedModule(referrer, specifier, payload, result) with the same result each time.

So it doesn't matter what the result is, what we are trying to validate is whether the dynamic import returns the same object each time. It doesn't matter for this test if it's an empty object, or an error, or whatever: the only thing we care is wether it's "the same result". Other tests ensures that dynamic imports resolve to the correct value, I don't see any value in adding that assertion to this test.